security/jwt/src/main/java/io/helidon/security/jwt/Jwt.java

```diff
@@ -663,11 +659,14 @@ public Optional<String> subject() {
 
     /**
      * User principal claim ("upn" from microprofile specification).
+     * Falls back "preferred_username" then "sub" claim.
      *
-     * @return user principal or empty if claim is not defined
+     * @return user principal or empty if claim and fallbacks are not defined
      */
     public Optional<String> userPrincipal() {
-        return userPrincipal;
+        return userPrincipal
```

Comment:
I think this causes inconsistency between the runtime model and the JWT which it generates. From my point of view this method should return only the set userPrincipal value and on places where this is used (and functionality based on the MP spec needs to be present), there might be those falbacks added. This causes backwards incompatible functionality, but on the other hand, I see this rather as a bug fix, so it should not be a problem.

Reply:
Any other opinions about this from others?